### ISSUE TITLE
Token help

### ISSUE THREAD

**Issue by rocrockyroc** [TheMrBooyah/hassio-dirigera-client#11]
Hello, 

Im reaching out to you for help with the dump file/token needed for Dirigera by ikea hub.

Im not familiar with the DUMP process. 
I am using RPi4 in home assistant 

if you can direct me to any further information, data or youtube videos, that be greatly appreciated to acquire this dump Token.

Regards

**Comment by TheMrBooyah** [comment]
#### Run the Dump Application

Prerequisites:
* Java 17 (or higher) has to be installed
  * Download Java the latest OpenJDK for your operating system [here](https://openjdk.org)
  * Unpack the zip file to a location of your choosing
  * Add `JAVA_HOME` to your environment variables with the following path `<PATH_TO_UNPACKED_ZIP_FILE>`
  * Add `%JAVA_HOME%\bin` to your path variable
  * Reboot your pc
  * Execute `java -version`
    * This should output something like:
    ```txt
      openjdk version "17.0.2" 2022-01-18
      OpenJDK Runtime Environment (build 17.0.2+8-86)
      OpenJDK 64-Bit Server VM (build 17.0.2+8-86, mixed mode, sharing)
    ```
* DIRIGERA Gateway is connected to the internet
  * Find the IP address of your DIRIGERA Gateway in your router.

##### Instructions
1) Download the latest [dirigera-client-dump.jar](https://github.com/dvdgeisler/DirigeraClient/releases)
2) Run `java -jar dirigera-client-dump.jar`</br>
   If the autodiscovery fails and no DIRIGERA is found in your Network, you may pass the DIRIGERA's connection details 
   by the parameters `--dirigera.hostname=<ip address of DIRIGERA Gateway>` and
   `--dirigera.port=8443`.

This will ask you to pair your gateway. After successfully pairing the gateway a file 'dirigera_access_token' will be
created. Open the file with your favourite text editor and copy everything into the 'Token' field.

**Comment by rocrockyroc** [comment]
Hello, thank you for the reply.  Today i spent some hours trying to get the token and learning Java.  The closest i have gotten was an error starting "No access token found. Pairing required"  this was dump file running in windows command prompt output.  It appears the dump program did find the hub. As the correct IP address etc were found.  
Apologies but Im very keen on this not familiar with terminologies etc.  If you have a further information or instruction please share with me.  Mean while I'm seeking help from a  java expert later this week.  I suspect its easy.

**Comment by TheMrBooyah** [comment]
If the pairing was successful you should have a file `dirigera_access_token` somewhere near where you ran the program. 
You can open this file with `notepad` and the string you see in the file is your access token.

I'm planning on making this program more user friendly since it's now made with the assumption people are technically inclined. I haven't had the time to do so just yet.

**Comment by rocrockyroc** [comment]
Hello,

Thank you for the reply.
I have the hub Paired with Apple homekit.  I did think maybe it needed pairing while software running but not sure how and what point if I need to press button on hub.

Below is output.  Perhaps might help you explain to me what I am doing wrong.

Thank you again

Rocco



C:\Users\roc_r\Downloads>java -jar dirigera-client-dump.jar

██████  ██ ██████  ██  ██████  ███████ ██████   █████       ██████ ██      ██ ███████ ███    ██ ████████
██   ██ ██ ██   ██ ██ ██       ██      ██   ██ ██   ██     ██      ██      ██ ██      ████   ██    ██
██   ██ ██ ██████  ██ ██   ███ █████   ██████  ███████     ██      ██      ██ █████   ██ ██  ██    ██
██   ██ ██ ██   ██ ██ ██    ██ ██      ██   ██ ██   ██     ██      ██      ██ ██      ██  ██ ██    ██
██████  ██ ██   ██ ██  ██████  ███████ ██   ██ ██   ██      ██████ ███████ ██ ███████ ██   ████    ██

2023-05-02 22:48:26.753  INFO 51496 --- [           main] d.d.i.d.client.dump.DumpApplication      : Starting DumpApplication v0.0.1-SNAPSHOT using Java 20.0.1 on Family-Desktop with PID 51496 (C:\Users\roc_r\Downloads\dirigera-client-dump.jar started by roc_r in C:\Users\roc_r\Downloads)
2023-05-02 22:48:26.753 DEBUG 51496 --- [           main] d.d.i.d.client.dump.DumpApplication      : Running with Spring Boot v2.7.5, Spring v5.3.23
2023-05-02 22:48:26.753  INFO 51496 --- [           main] d.d.i.d.client.dump.DumpApplication      : No active profile set, falling back to 1 default profile: "default"
2023-05-02 22:48:27.159  INFO 51496 --- [           main] d.d.i.d.c.api.http.GatewayDiscovery      : Auto discover gateway
2023-05-02 22:48:27.163  INFO 51496 --- [           main] d.d.i.d.client.api.http.TokenStore       : Load access token
2023-05-02 22:48:27.163 ERROR 51496 --- [           main] d.d.i.d.client.api.http.TokenStore       : Cannot read access token: dirigera_access_token (The system cannot find the file specified)
2023-05-02 22:48:27.432  INFO 51496 --- [pool-1-thread-1] d.d.i.d.c.api.http.GatewayDiscovery      : Gateway discovered: name=gw2-831636084ee8, server=gw2-831636084ee8._ihsp._tcp.local., application=ihsp, protocol=tcp, domain=local
2023-05-02 22:48:27.434  INFO 51496 --- [pool-1-thread-1] d.d.i.d.c.api.http.GatewayDiscovery      : Gateway rest API discovered: name=gw2-831636084ee8, port=8443, ip=192.168.0.84, server=gw2-831636084ee8.local., application=ihsp, protocol=tcp, domain=local
2023-05-02 22:48:27.537  INFO 51496 --- [           main] d.d.i.d.client.api.http.ClientOAuthApi   : Dirigera client name: Family-Desktop
2023-05-02 22:48:27.610  INFO 51496 --- [           main] d.d.i.d.client.dump.DumpApplication      : Started DumpApplication in 1.133 seconds (JVM running for 1.4)
2023-05-02 22:48:27.611  INFO 51496 --- [           main] d.d.i.d.client.api.http.ClientOAuthApi   : No access token found. Pairing required.
2023-05-02 22:48:27.945 ERROR 51496 --- [           main] o.s.boot.SpringApplication               : Application run failed

java.lang.IllegalStateException: Failed to execute CommandLineRunner
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:774) ~[spring-boot-2.7.5.jar!/:2.7.5]
        at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:755) ~[spring-boot-2.7.5.jar!/:2.7.5]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:315) ~[spring-boot-2.7.5.jar!/:2.7.5]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1306) ~[spring-boot-2.7.5.jar!/:2.7.5]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1295) ~[spring-boot-2.7.5.jar!/:2.7.5]
        at de.dvdgeisler.iot.dirigera.client.dump.DumpApplication.main(DumpApplication.java:60) ~[classes!/:0.0.1-SNAPSHOT]
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[na:na]
        at java.base/java.lang.reflect.Method.invoke(Method.java:578) ~[na:na]
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49) ~[dirigera-client-dump.jar:0.0.1-SNAPSHOT]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:108) ~[dirigera-client-dump.jar:0.0.1-SNAPSHOT]        at org.springframework.boot.loader.Launcher.launch(Launcher.java:58) ~[dirigera-client-dump.jar:0.0.1-SNAPSHOT]
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65) ~[dirigera-client-dump.jar:0.0.1-SNAPSHOT]
Caused by: org.springframework.web.reactive.function.client.WebClientRequestException: Failed to resolve 'gw2-831636084ee8' after 3 queries ; nested exception is java.net.UnknownHostException: Failed to resolve 'gw2-831636084ee8' after 3 queries
        at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:141) ~[spring-webflux-5.3.23.jar!/:5.3.23]
        Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException:
Error has been observed at the following site(s):
        *__checkpoint Ôçó Request to GET https://gw2-831636084ee8:8443/v1/oauth/authorize?audience=homesmart.local&response_type=code&code_challenge=k2ribJfPcnftRcWuAiiKDm84cDcK7nHGYf7PeXIXqxw&code_challenge_method=S256 [DefaultWebClient]
Original Stack Trace:
                at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:141) ~[spring-webflux-5.3.23.jar!/:5.3.23]
                at reactor.core.publisher.MonoErrorSupplied.subscribe(MonoErrorSupplied.java:55) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.Mono.subscribe(Mono.java:4455) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.MonoNext$NextSubscriber.onError(MonoNext.java:93) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onError(MonoFlatMapMany.java:204) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.SerializedSubscriber.onError(SerializedSubscriber.java:124) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.whenError(FluxRetryWhen.java:225) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.FluxRetryWhen$RetryWhenOtherSubscriber.onError(FluxRetryWhen.java:274) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.drain(FluxConcatMap.java:415) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.onNext(FluxConcatMap.java:251) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.EmitterProcessor.drain(EmitterProcessor.java:537) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.EmitterProcessor.tryEmitNext(EmitterProcessor.java:343) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.SinkManySerialized.tryEmitNext(SinkManySerialized.java:100) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.InternalManySink.emitNext(InternalManySink.java:27) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.onError(FluxRetryWhen.java:190) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:201) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.netty.http.client.HttpClientConnect$MonoHttpConnect$ClientTransportSubscriber.onError(HttpClientConnect.java:307) ~[reactor-netty-http-1.0.24.jar!/:1.0.24]
                at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:201) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.netty.resources.DefaultPooledConnectionProvider$DisposableAcquire.onError(DefaultPooledConnectionProvider.java:158) ~[reactor-netty-core-1.0.24.jar!/:1.0.24]
                at reactor.netty.internal.shaded.reactor.pool.AbstractPool$Borrower.fail(AbstractPool.java:475) ~[reactor-netty-core-1.0.24.jar!/:1.0.24]
                at reactor.netty.internal.shaded.reactor.pool.SimpleDequePool.lambda$drainLoop$9(SimpleDequePool.java:431) ~[reactor-netty-core-1.0.24.jar!/:1.0.24]
                at reactor.core.publisher.FluxDoOnEach$DoOnEachSubscriber.onError(FluxDoOnEach.java:186) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:201) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.netty.resources.DefaultPooledConnectionProvider$PooledConnectionAllocator$PooledConnectionInitializer.onError(DefaultPooledConnectionProvider.java:548) ~[reactor-netty-core-1.0.24.jar!/:1.0.24]
                at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:192) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:259) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:106) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.Operators.error(Operators.java:198) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.MonoError.subscribe(MonoError.java:53) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.Mono.subscribe(Mono.java:4455) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.netty.transport.TransportConnector$MonoChannelPromise.tryFailure(TransportConnector.java:538) ~[reactor-netty-core-1.0.24.jar!/:1.0.24]
                at reactor.netty.transport.TransportConnector.lambda$doResolveAndConnect$11(TransportConnector.java:345) ~[reactor-netty-core-1.0.24.jar!/:1.0.24]
                at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:552) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:609) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.setFailure(DefaultPromise.java:109) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.resolver.InetSocketAddressResolver$2.operationComplete(InetSocketAddressResolver.java:86) ~[netty-resolver-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:609) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.resolver.dns.DnsResolveContext.finishResolve(DnsResolveContext.java:1055) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.resolver.dns.DnsResolveContext.tryToFinishResolve(DnsResolveContext.java:1000) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.resolver.dns.DnsResolveContext.query(DnsResolveContext.java:418) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.resolver.dns.DnsResolveContext.onResponse(DnsResolveContext.java:629) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.resolver.dns.DnsResolveContext.access$400(DnsResolveContext.java:66) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.resolver.dns.DnsResolveContext$2.operationComplete(DnsResolveContext.java:462) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:605) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.resolver.dns.DnsQueryContext.trySuccess(DnsQueryContext.java:232) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.resolver.dns.DnsQueryContext.finish(DnsQueryContext.java:224) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.resolver.dns.DnsNameResolver$DnsResponseHandler.channelRead(DnsNameResolver.java:1314) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[netty-codec-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.channel.nio.AbstractNioMessageChannel$NioMessageUnsafe.read(AbstractNioMessageChannel.java:97) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
                at java.base/java.lang.Thread.run(Thread.java:1623) ~[na:na]
        Suppressed: java.lang.Exception: #block terminated with an error
                at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:99) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at reactor.core.publisher.Mono.block(Mono.java:1707) ~[reactor-core-3.4.24.jar!/:3.4.24]
                at de.dvdgeisler.iot.dirigera.client.dump.DumpApplication.lambda$run$1(DumpApplication.java:34) ~[classes!/:0.0.1-SNAPSHOT]
                at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:771) ~[spring-boot-2.7.5.jar!/:2.7.5]
                at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:755) ~[spring-boot-2.7.5.jar!/:2.7.5]
                at org.springframework.boot.SpringApplication.run(SpringApplication.java:315) ~[spring-boot-2.7.5.jar!/:2.7.5]
                at org.springframework.boot.SpringApplication.run(SpringApplication.java:1306) ~[spring-boot-2.7.5.jar!/:2.7.5]
                at org.springframework.boot.SpringApplication.run(SpringApplication.java:1295) ~[spring-boot-2.7.5.jar!/:2.7.5]
                at de.dvdgeisler.iot.dirigera.client.dump.DumpApplication.main(DumpApplication.java:60) ~[classes!/:0.0.1-SNAPSHOT]
                at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[na:na]
                at java.base/java.lang.reflect.Method.invoke(Method.java:578) ~[na:na]
                at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49) ~[dirigera-client-dump.jar:0.0.1-SNAPSHOT]
                at org.springframework.boot.loader.Launcher.launch(Launcher.java:108) ~[dirigera-client-dump.jar:0.0.1-SNAPSHOT]
                at org.springframework.boot.loader.Launcher.launch(Launcher.java:58) ~[dirigera-client-dump.jar:0.0.1-SNAPSHOT]
                at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65) ~[dirigera-client-dump.jar:0.0.1-SNAPSHOT]
Caused by: java.net.UnknownHostException: Failed to resolve 'gw2-831636084ee8' after 3 queries
        at io.netty.resolver.dns.DnsResolveContext.finishResolve(DnsResolveContext.java:1047) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.resolver.dns.DnsResolveContext.tryToFinishResolve(DnsResolveContext.java:1000) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.resolver.dns.DnsResolveContext.query(DnsResolveContext.java:418) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.resolver.dns.DnsResolveContext.onResponse(DnsResolveContext.java:629) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.resolver.dns.DnsResolveContext.access$400(DnsResolveContext.java:66) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.resolver.dns.DnsResolveContext$2.operationComplete(DnsResolveContext.java:462) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:605) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.resolver.dns.DnsQueryContext.trySuccess(DnsQueryContext.java:232) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.resolver.dns.DnsQueryContext.finish(DnsQueryContext.java:224) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.resolver.dns.DnsNameResolver$DnsResponseHandler.channelRead(DnsNameResolver.java:1314) ~[netty-resolver-dns-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[netty-codec-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.channel.nio.AbstractNioMessageChannel$NioMessageUnsafe.read(AbstractNioMessageChannel.java:97) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[netty-transport-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.84.Final.jar!/:4.1.84.Final]
        at java.base/java.lang.Thread.run(Thread.java:1623) ~[na:na]

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by TheMrBooyah** [comment]
It seems the application tries to authenticate itself for pairing but isn't able to resolve the hostname of the hub to an IP.
Try running the command `java -jar dirigera-client-dump.jar --dirigera.hostname=192.168.0.84 --dirigera.port=8443`.

Please post the results in case anything still goes wrong

**Comment by rocrockyroc** [comment]
Hello,
Thank you for your help.  The information helped and it worked. Once it worked it asked for Pairing Button for which I wasn’t sure where that was.
Ikea do not have data or instruction on “Pairing button”  Once I worked it out I had to press this for about 3 seconds then it works. Quick press didn’t work.

Also once everything was going I had few errors.  I had to reboot HA and last re start the this app.

All good now.  Hope this helps  others too.

Thank you again



Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows
